### PR TITLE
Bug/2963 search shows no results after switching to another tool and switching back to debugger

### PR DESCRIPTION
**Pull request by codehag**
Associated Issue: #2963

### CODE REVIEW

File: src/components/Editor/SearchBar.js [hunk 1]
```diff
@@ -697,8 +701,9 @@ class SearchBar extends Component {
   }
 
   renderResults() {
-    const { symbolSearchResults, selectedResultIndex } = this.state;
-    const { query, symbolSearchOn } = this.props;
+    const { selectedResultIndex } = this.state;
+    const { query, symbolSearchResults, symbolSearchOn } = this.props;
+    console.log("hi");
```
Comment (jasonLaster) [inline src/components/Editor/SearchBar.js]: 👋

File: src/reducers/tests/ui.js [hunk 2]
```diff
@@ -28,4 +28,16 @@ describe("ui reducer", () => {
...
+  it("gets the search results", () => {
+    const state = State();
+    expect(getSearchResults({ ui: state })).toBe(state.searchResults);
+  });
+
+  it("gets the symbol search results", () => {
+    const state = State();
+    expect(getSymbolSearchResults({ ui: state })).toBe(
+      state.symbolSearchResults
+    );
```
Comment (jasonLaster) [inline src/reducers/tests/ui.js]: I like treating actions as the public interface here:

here's an actions/tests/ui test
```js
  it("should toggle the visible state of project search", () => {
    const { dispatch, getState } = createStore();
    expect(getProjectSearchState(getState())).toBe(false);
    dispatch(actions.toggleProjectSearch());
    expect(getProjectSearchState(getState())).toBe(true);
  });
```

File: src/components/Editor/SearchBar.js (hunk 1 above)
Comment (codehag) [inline src/components/Editor/SearchBar.js]: ack

File: src/reducers/tests/ui.js (hunk 2 above)
Comment (codehag) [inline src/reducers/tests/ui.js]: I guess false / true should be replaced with something else? or maybe we drop these tests for now?

Comment (jasonLaster) [inline src/reducers/tests/ui.js]: oh, i think this is a good "model" test

**Pull request merged**